src/Stryker.Core/Stryker.Core/DashboardCompare/GitInfoProvider.cs

```diff
@@ -128,17 +102,39 @@ private Commit GetCommit()
                 return sourceBranch.Tip;
             }
 
-            if (_options.GitSource.Length == 40)
+            // It's a commit!
+            if (_options.GitDiffTarget.Length == 40)
             {
-                var commit = Repository.Lookup(new ObjectId(_options.GitSource)) as Commit;
+                var commit = Repository.Lookup(new ObjectId(_options.GitDiffTarget)) as Commit;
 
                 if (commit != null)
                 {
+                    _logger.LogDebug($"Found commit {commit.Sha} for branch {_options.GitDiffTarget}");
                     return commit;
                 }
             }
 
             return null;
         }
+
+        private void CreateLocalBranchForGitDiffTarget()
```

Comment:
This was added to be able to determine the current branch in a CI/CD scenario, but we can clearly see that it's not working. I'm wondering if we even need it or if we can rather just use `Repository.Head.Tip` as the commit to compare to. @Stijn-Rutten what do you think?

Reply:
@Mobrockers in PR scénario if we focus only on the diff feature we should only indeed compare repository.head.tip to the target branch of the pull request. Thus we don't need to create temp branch and we should ignore index. I propose that we first fix this feature and then have a look to the dashboard feature. To ease fixing the diff feature the stryker on stryker task should only use the --diff cmd line arg. Once in an azure devops PR we see that stryker only mutates the same files as the one detected as changed in azure devops we should investigate the dashboard compare feature.